### PR TITLE
ER-1156 - Styles and JS added to library

### PR DESCRIPTION
**Pull request by c0ntax**

### CODE REVIEW

**Comment by DavidHughes** [comment]
LGTM - will let @dgcollard merge as he's assigned it.

**Comment by dgcollard** [comment]
:bow:

File: assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+jQuery( document ).ready(function( $ ) {
+    'use strict';
+    // First, apply to any pre-checked options
+    $('form').find('input[type=radio]:checked, input[type=checkbox]:checked').each(function() {
+        checkIt($(this));
+    });
+
+    // Now set up listeners to ensure that any changes are also checked
+    $('form').on('change', 'input[type=radio], input[type=checkbox]', function() {
```
Comment (heylookalive) [inline assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js]: Could we not do the following avoiding an anon function?
`$('form').on('change', 'input[type=radio], input[type=checkbox]', checkIt);`

**Comment by heylookalive** [comment]
Otherwise looks great 😄

File: assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js (hunk 1 above)
Comment (c0ntax) [inline assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js]: Does that pass through 'this'?

**Comment by dgcollard** [comment]
¯\\\_(ツ)\_/¯

File: assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js (hunk 1 above)
Comment (DavidHughes) [inline assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js]: Yep, it should pass it through. Good shout.

Comment (c0ntax) [inline assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js]: I will affect change in a new pull request given that this one has been merged. ;-)

Comment (c0ntax) [inline assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js]: Hmm.. The problem is that checkIt() runs via two different ways. Once is triggered by an event (and therefore the Event object is passed to checkIt when not using anonymous functions) and the other is when the page first loads and we look to set all default values which aren't event driven.

So checkIt would have to accept two different data types. It seems tidier to me to have the anonymous function pass in the correct data.

Comment (heylookalive) [inline assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js]: I think $(this) is supplied to the callback without needing a function parameter to be defined. That way you could have `checkIt(){` and then just use `$(this)` which represents the element. No idea how this actually works...

Comment (c0ntax) [inline assets/js/cruk-base/plugin-radio-and-checkboxes-checked.js]: Just tried it out. It wasn't passed.

Hmm... Maybe checkIt()(this) will work? Will investigate in a second.

**Pull request merged**